delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+
+import six
+
+from oslo_log import log as logging
+
+from delfin import exception, utils
+from delfin.common import constants
+from delfin.drivers.utils.ssh_client import SSHPool
+from delfin.drivers.hitachi.hnas import constants as constant
+from delfin.drivers.utils.tools import Tools
+
+LOG = logging.getLogger(__name__)
+
+
+class NasHandler(object):
+
+    def __init__(self, **kwargs):
+        self.ssh_pool = SSHPool(**kwargs)
+        self.evs_list = []
+
+    @staticmethod
+    def get_size(limit, is_calculate=False):
+        if limit == '0B':
+            return 0
+        if limit == '-':
+            return 0 if is_calculate else '-'
+        return int(Tools.get_capacity_size(limit))
+
+    def ssh_do_exec(self, command_list):
+        res = ''
+        res = self.ssh_pool.do_exec_command(command_list)
+        while 'Failed to establish SSC connection' in res:
+            res = self.ssh_pool.do_exec_command(command_list)
+        return res
+
+    def login(self):
+        try:
+            result = self.ssh_do_exec(['cluster-show -y'])
+            if 'is not a recognized command' in result \
+                    or 'Unknown command' in result \
+                    or 'EVS' not in result:
+                raise exception.InvalidIpOrPort()
+        except Exception as e:
+            LOG.error("Failed to login netapp %s" %
+                      (six.text_type(e)))
+            raise e
+
+    @staticmethod
+    def split_value_map_list(
+            value_info,
+            map_list,
+            value_key,
+            line='\r\n',
+            split=":",
+            split_key=None):
+        detail_array = value_info.split(line)
+        value_map = {}
+        for detail in detail_array:
+            if detail:
+                string_info = detail.split(split)
+                key = string_info[0].replace(' ', '')
+                value = ''
+                if len(string_info) > 1:
+                    for string in string_info[1:]:
+                        value += string.\
+                            replace('""', '').\
+                            replace('\'', '').\
+                            replace(' ', '')
+                if value_map.get(key):
+                    value_map[key + '1'] = value
+                else:
+                    value_map[key] = value
+            else:
+                if value_map != {} and value_key in value_map:
+                    map_list.append(value_map)
+                value_map = {}
+            if split_key and split_key in detail:
+                if value_map != {} and value_key in value_map:
+                    map_list.append(value_map)
+                value_map = {}
+        if value_map != {} and value_key in value_map:
+            map_list.append(value_map)
+
+    @staticmethod
+    def get_table_data(values):
+        header_index = 0
+        table = values.split("\r\n")
+        for i in range(0, len(table)):
+            if constant.PATTERN.search(table[i]):
```

Comment:
What's this `PATTERN` for? Please use a meaningful name.

Reply:
In order to search out the header of the data table and delete the part above the header